script/echoserver.py

```diff
@@ -0,0 +1,54 @@
+import socket
+import sys
+
+servport = 1234
+proxyport = 1111
```

Comment:
What is **proxyport** for? It looks that it is used in the same way as **servport**. And, clientecho.py only connect **serverport**

Reply:
in the 'tproxy_iptables.sh' setup script the following lines

iptables -t mangle -A PREROUTING -p tcp --dport $ser_port -j TPROXY \
--tproxy-mark $mark/$mark --on-point $lst_port

Allows us to not only intercept the client connection at the proxy server, but also redirect them to a different port number. This gives us more flexibility in the proxy server since we can have our application listening on any port number, not just the port number the client is trying to connect to. In this example, the client is trying to connect to the server on port 1234, but the proxy server can redirect it to its own port 1111 where the python script is listening to if it's run as a proxy.